src/pages/instances/forms/UploadExternalFormatFileForm.tsx

```diff
@@ -0,0 +1,333 @@
+      .then((operation) => {
+        const operationId = operation.metadata.id;
+        const operationSecret = operation.metadata?.metadata?.fs;
+        const instanceName = getInstanceName(operation.metadata);
+
+        // establish websocket connection based on the instance creation operation
+        let wsUrl = `wss://${location.host}/1.0/operations/${operationId}/websocket`;
+        if (operationSecret) {
+          wsUrl += `?secret=${operationSecret}`;
+        }
+
+        const ws = new WebSocket(wsUrl);
+
+        ws.onopen = () => {
+          sendFileByWebSocket(
+            ws,
+            formik.values.imageFile,
+            FILE_CHUNK_SIZE,
+            handleUploadProgress.bind(null, instanceName),
+            handleUploadError,
+          );
+        };
+
+        // set up event queue for the operation
+        eventQueue.set(
+          operationId,
+          () => handleSuccess(getInstanceName(operation.metadata)),
+          handleFailure,
+          invalidateCache,
+        );
+      })
+      .catch((e) => {
+        close();
+        toastNotify.failure("Instance creation failed.", e);
+      });
+  };
+
+  const formik = useFormik<UploadExternalFormatFileFormValues>({
+    initialValues: {
+      name: "",
+      pool: "",
+      imageFile: null,
+      formatConversion: true,
+      virtioConversion: false,
+      architecture: supportedVMArchOptions(
+        settings?.environment?.architectures || [],
+      )[0].value,
+    },
+    validateOnMount: true,
+    validationSchema: Yup.object().shape({
+      name: instanceNameValidation(
+        project?.name || "",
+        instanceNameAbort,
+      ).optional(),
+    }),
+    onSubmit: handleConfirmUpload,
+  });
+
+  const handleFileChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("imageFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      const suffix = "-import";
+      const fileExtension = getFileExtension(file.name);
+      const sanitisedFileName = sanitizeInstanceName(
+        file.name.split(fileExtension)[0],
+      );
+      const instanceName = truncateInstanceName(sanitisedFileName, suffix);
+      await formik.setFieldValue("imageFile", file);
+      await formik.setFieldValue("name", instanceName);
+
+      // validate instance name
+      await formik.validateField("name");
+      void formik.setFieldTouched("name", true, true);
+      if (!formik.errors.name) {
+        formik.setFieldError("name", undefined);
+      }
+
+      // If the image is already in raw format, remove the format conversion option
+      // this will optimise the conversion process since raw vm images do not need to be converted
+      const isRawImage = await isImageTypeRaw(file);
+      setIsRawImage(isRawImage);
+      if (isRawImage) {
+        await formik.setFieldValue("formatConversion", false);
+      }
```

Comment:
I don't fully understand why we set this field and then disable the selector for it.

Should we maybe only set the field to false and remove the `isRawImage` state and its side effect to disable the field? Disabling it seems rather drastic to me and might be experienced as not wanted by users.

Reply:
Yeah that makes sense, I guess the tooltip also indicates that the option can be skipped if it's a raw image, will remove the disabling logic.